analyzer/src/main/kotlin/managers/utils/NuGetSupport.kt

```diff
@@ -111,18 +104,23 @@ class NuGetSupport(serviceIndexUrls: List<String> = listOf(DEFAULT_SERVICE_INDEX
         }
     }
 
-    private val serviceIndices = runBlocking {
-        serviceIndexUrls.map {
-            async { JSON_MAPPER.readValueFromUrl<ServiceIndex>(it) }
-        }.awaitAll()
-    }
+    private val registrationsBaseUrls: List<String> = run {
```

Comment:
Maybe make this `by lazy {}` now?

Reply:
This is out of scope and won't be needed anyway, because it will become lazy even without the `lazy` operator in a following change.

Reply:
> This is out of scope

You're now moving more (IO-related) code here, making it potentially more expensive to run, so I don't see how in turn making it lazy would be out of scope, just because it slightly changes the behavior of the property.

In general I'm not very happy with this commit on its own as it mangles getting the base URLs with reading from them. But I get that there's not much point in debating about it, as you're changing the code later anyway.

Reply:
> You're now moving more (IO-related) code here, making it potentially more expensive to run, 

To my understanding that's incorrect. Prior and after this change the exact same logic is executed during construction time.